Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3621,8 +3621,10 @@ void RenderBox::computeBlockDirectionMargins(const RenderBlock& containingBlock,
     // the containing block (8.3)
     LayoutUnit cw = containingBlockLogicalWidthForContent();
     const RenderStyle& containingBlockStyle = containingBlock.style();
-    marginBefore = minimumValueForLength(style().marginBeforeUsing(&containingBlockStyle), cw);
-    marginAfter = minimumValueForLength(style().marginAfterUsing(&containingBlockStyle), cw);
+    if (!containingBlock.shouldTrimChildMargin(MarginTrimType::BlockStart, *this))
+        marginBefore = minimumValueForLength(style().marginBeforeUsing(&containingBlockStyle), cw);
+    if (!containingBlock.shouldTrimChildMargin(MarginTrimType::BlockEnd, *this))
+        marginAfter = minimumValueForLength(style().marginAfterUsing(&containingBlockStyle), cw);
```

Comment:
Should `marginBefore`/`marginAfter` be initialized to anything if we are doing a trim?